Update platform support section in module-loader.md (#7)
openmoq/moqx: design/module-loader.md

### Platform Support

> This plugin system uses `dlopen`/`dlsym` and works on systems that support these APIs (Linux, macOS, other POSIX systems). Windows is not currently supported. Multiple `.so` or those that have dependencies on other `.so` is disallowed on macOS; this will require special handling when macOS is the target OS.

Reviewer: This is a great note, but I am not sure about the goal here. Are we pushing the solution down the line? 

I think we should either
1. Only do Linux support (per [the spec](https://github.com/OpenMOQ/relay-requirements/blob/f9e11d6ebfc6298458b7a3f8f554b0e4ec9a6785/relay-requirements.md#L87))
2. or do a proper non-linux support at every step. 

I would push for linux-only support. Generally there will be more places where those differences between linux/posix/bsd/macos pop-up. Doing abstractions above OS differences is sometimes a lot of work and often hard not to lose advantages of one of the systems (e.g. io_uring). 

I do not think it makes sense to support macOS just because developers use it (including me). Will anybody run this server-grade relay on macOS? I would not think so. It would make more sense to me to support BSD if somebody needs that; see https://github.com/OpenMOQ/relay-requirements/issues/8. 

I am happy to help people with VMs/containers dev setup on macOS. 

I am not going to block non-linux in principle alone, but we need to put the work there. I do not really have experience with non-Linux high performance server stuff. Can somebody describe how the Mac solution is supposed to work here? What's the special handling required? The rest of the documents talks about loading multiple .so's, how is this supposed to work on macos? 

-----

Sorry to bikeshed on a short PR, but I would like to resolve this question generically, and this short PR seems like a fitting place.

Reviewer 2: I think we can say mac support is a low-pri side goal, and features like custom module loading might be self-supported.

Author: Yes, I understood this as Linux only, but if we also support local builds for quick dev testing, the option of a macos build would be a nice-to-have for the 90% of people on the Team using macos (I am not one of them, but I understand the need).
<!-- Reviewable comment -OngeCkY5nNE7MalQnXx:b1vurvk -->
<!-- Sent from Reviewable.io -->

Reviewer: I am sorry, but I don't feel my arguments are addressed in the response. 
<!-- Reviewable comment -OoWgZL42gDsFNWJHWvQ:byo0pb5 -->
<!-- Sent from Reviewable.io -->

Author: Lets just remove it; the purpose of this was an informational note; it's not important enough to keep IMHO.
<!-- Reviewable comment -Oo_M3Fd3cdAsPsX2vEB:b-rhwf5c -->
<!-- Sent from Reviewable.io -->

> This plugin system uses `dlopen`/`dlsym` and works on systems that support these APIs (Linux, macOS, other POSIX systems). Windows is not currently supported. Multiple `.so` or those that have dependencies on other `.so` is disallowed on macOS; this will require special handling when macOS is the target OS.

Reviewer 2: ```suggestion
This plugin system uses `dlopen`/`dlsym` and works on Linux. Windows and macOS builds will not support plugins initially.
```
